github/github.go

```diff
@@ -575,8 +574,7 @@ func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Res
 			decErr := json.NewDecoder(resp.Body).Decode(v)
 			if decErr == io.EOF {
 				decErr = nil // ignore EOF errors caused by empty response body
-			}
-			if decErr != nil {
+			} else if decErr != nil {
```

Comment:
Actually, lines 574-579 could be simplified to this, which I think is clearer:

```go
if decErr := json.NewDecoder(resp.Body).Decode(v); decErr != io.EOF {
  err = decErr  // ignore EOF errors caused by empty response body
}
```

But I'm betting that the original code review wanted to make an explicit case for `io.EOF` and wanted to keep the logic super-easy to read...

So I'm leaning toward closing this entire PR since the first part is not correct, and this change is not needed.

Reply:
I agree with you it's important to keep the logic super-easy to read. I meant with this change only removing the extra execution of `decErr != nil`. So if the value of  `decErr` is  `io.EOF`, there is no need to do again `decErr != nil` because for sure it will never be `nill`.

Reply:
I understand, but my point was that sometimes we decide to waste a CPU cycle or two just for the sake of readability of the code, and I would have to go back to the original code review where these lines were discussed to find out if that was the case here.

Reply:
I see. Thank you for the clarification, it makes sense.